docs/defi/chain-key-tokens/ckbtc/using-ckbtc-in-dapps.mdx

```diff
@@ -10,172 +10,89 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+## Ledger interaction
+
+The [ckBTC ledger](https://dashboard.internetcomputer.org/canister/mxzaz-hqaaa-aaaar-qaada-cai) implements the [ICRC-1](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-1/README.md) and [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) token standards.
+This means that integration with the ledger involves calling the standard endpoints for actions such as:
+
+- `icrc1_balance_of` to fetch balances,  
+- `icrc1_transfer` to transfer tokens, and
+- `icrc2_approve` / `icrc2_transfer_from` for delegated transfers.
+
+There are two primary ways to interact with the ckBTC ledger:
+
+### Client calls (via agent)
+A client application (e.g. web application) can use an [agent](/docs/building-apps/interact-with-canisters/agents/overview) to interact directly with the ckBTC ledger canister.
+
+An **agent** is the API client layer used to call ICP canisters. For example, the [ICP JavaScript agent](/docs/building-apps/interact-with-canisters/agents/javascript-agent) can create an actor from the ledger’s [Candid interface](/docs/building-apps/interact-with-canisters/candid/candid-concepts) to invoke its endpoints.
+
+The following libraries simplify creating actors for the ckBTC ledger:
+
+- Rust: [icrc-ledger-agent](https://crates.io/crates/icrc-ledger-agent)
+- TypeScript: [@dfinity/ledger-icrc](https://www.npmjs.com/package/@dfinity/ledger-icrc)
+
+In this model, all transactions are executed from the user’s own [account](/docs/defi/token-standards/#accounts-principals-and-subaccounts) (typically the principal without a subaccount).
+
+### Inter-canister calls
+A canister can also interact with the ckBTC ledger.
+
+- In some cases, the canister may use ckBTC for its own purposes, such as making payments to other canisters, managing a treasury, or executing automated transfers.
+- In other cases, the canister may act on behalf of users, which requires managing user funds. This can be achieved by creating **subaccounts per user** or by **pooling all funds in a single account** and maintaining an internal balance table.
+  - It is recommended to use the [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) approval flow (`icrc2_approve` / `icrc2_transfer_from`) to move funds from a user’s wallet to the backend canister.
+
+Libraries are available to simplify inter-canister calls:
+
+- Rust: [icrc-ledger-client-cdk](https://crates.io/crates/icrc-ledger-client-cdk)
+- Motoko: [icrc2-types](https://mops.one/icrc2-types) (includes ICRC-1 types)
+
+Example code:
+- Motoko: [transfer](https://github.com/dfinity/examples/tree/master/motoko/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/motoko/token_transfer_from)
+- Rust: [transfer](https://github.com/dfinity/examples/tree/master/rust/token_transfer), [transfer_from](https://github.com/dfinity/examples/tree/master/rust/token_transfer_from)
```

Comment:
what about this example?
https://github.com/dfinity/examples/tree/master/rust/token_transfer_from

Reply:
this is actually there 🤔